docs/tutorial/composition.md

```diff
@@ -0,0 +1,219 @@
+# Composition
+
+Starting with version 2.5.0, `concrete-python` supports circuit __composition__ ! Composition allows you to use the output of a circuit execution, directly as an input, without the need to decrypt it in between. You can execute the circuit as many time as you want, by forwarding outputs, without decrypting intermediate values. This feature enables a new range of applications among which, a support for iterations with cleartext iteration control.
+
+Here is a first simple example that uses composition to implement a simple counter in fhe:
+
+```python
+from concrete import fhe
+
+@fhe.compiler({"counter": "encrypted"})
+def increment(counter):
+   return (counter + 1) % 100
+
+print("Compiling `increment` function")
+increment_fhe = increment.compile(list(range(0, 100)), composable=True)
+
+print("Generating keyset ...")
+increment_fhe.keygen()
+
+print("Encrypting the initial counter value")
+counter = 0
+counter_enc = increment_fhe.encrypt(counter)
+
+print(f"| iteration || decrypted | cleartext |")
+for i in range(10):
+    counter_enc = increment_fhe.run(counter_enc)
+    counter = increment(counter)
+
+    # For demo purpose; no decryption is needed.
+    counter_dec = increment_fhe.decrypt(counter_enc)
+    print(f"|     {i}     || {counter_dec:<9} | {counter:<9} |")
+```
+
+Note the use of the `composable` flag in the `compile` call. It instructs the compiler to ensure the crypto-paramters and fhe circuit generation are so that calling the circuit in a composed way is correct. Hopefully, you don't have to worry about those details; composition works out of the box. Executing this script should give the following output:
```

Comment:
Same remark about long line.
Don't mention the details if they don't have to worry about them. (I think they can be added after, i.e. extra cost, extra precision as they mostly apply to multi input/ multi output)

It guarantees that the circuit can be called on its own outputs. See Limitations for more details.

Reply:
Done